src/simulations/SimulationFactory.cpp

```diff
@@ -12,8 +12,29 @@ std::unique_ptr<Simulation> SimulationFactory::createSimulation(SimulationType t
     case SimulationType::LJ:
         SPDLOG_DEBUG("Generating LJ...");
         return std::make_unique<LennardJones>(filename, args);
+    case SimulationType::LJLC:
+        SPDLOG_DEBUG("Generating LJLC...");
+        return std::make_unique<LennardJonesLC>(filename, args);
     default:
         CLIUtils::error("Invalid Simulation type!", "", false);
     }
     return std::make_unique<Gravity>(filename, args);
+}
+
+std::unique_ptr<Simulation> SimulationFactory::createSimulation(SimulationType type, ParticleContainer &pc,
+                                                                const Arguments &args) {
+    switch (type) {
+    case SimulationType::GRAVITY:
+        SPDLOG_DEBUG("Generating Gravity...");
+        return std::make_unique<Gravity>(pc, args);
+    case SimulationType::LJ:
+        SPDLOG_DEBUG("Generating LJ...");
+        return std::make_unique<LennardJones>(pc, args);
+    case SimulationType::LJLC:
+        SPDLOG_DEBUG("Generating LJLC...");
+        return std::make_unique<LennardJonesLC>(pc, args);
+    default:
+        CLIUtils::error("Invalid Simulation type!", "", false);
+    }
```

Comment:
This brings me back to the structure of the code, if you want to do LinkedCells with gravity (while not required for this course, so you don't need to implement), you will have to implement another simulation subtype. Ideally, this should be possible right away as you have both the container and force available, so something like ```Simulation<ContainterType, ForceType>()```

Reply:
This is something we'll implement soon in the new `Assignment4` branch, since it's quite a large change. We've decided to refactor the simulation structure completely. We'll get rid of the individual simulation classes in favor of one overarching `Simulation` template class (as you suggested).